File: src/main/java/boj/problems/step19/No10828.java
package boj.problems.step19;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class No10828 {
    static class Stack {
        int size = 0;
        int[] arr;

        Stack(int stackSize) {
            this.arr = new int[stackSize];
        }

        public void push(int data) {
            this.arr[size] = data;
            size++;
        }

        public int pop() {
            if (this.size == 0) {
                return -1;
            }

            int ret = this.arr[this.size - 1];
            this.arr[this.size - 1] = 0;
            this.size--;

            return ret;
        }

        public int size() {
            return this.size;
        }

        public int empty() {
            return this.size == 0 ? 1 : 0;
        }

        public int top() {
            if (this.size == 0) {
                return -1;
            }

            return this.arr[this.size - 1];
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter output = new BufferedWriter(new OutputStreamWriter(System.out));

        solve(input, output);

        output.flush();
        output.close();
    }

    static void solve(BufferedReader input, BufferedWriter output) throws IOException {
        int n = Integer.parseInt(input.readLine());

        Stack stack = new Stack(n);

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(input.readLine(), " ");

            switch (st.nextToken()) {
                case "push":
                    stack.push(Integer.parseInt(st.nextToken()));
                    break;
                case "pop":
                    output.write(stack.pop() + "\n");
                    break;
                case "size":
                    output.write(stack.size() + "\n");
                    break;
                case "empty":
                    output.write(stack.empty() + "\n");
                    break;
                case "top":
                    output.write(stack.top() + "\n");
                    break;
            }
        }
    }
}

File: src/main/kotlin/boj/problems/step19/No10828.kt
package boj.problems.step19

import java.io.BufferedReader
import java.io.IOException
import java.util.StringTokenizer

class No10828 {
    @Throws(IOException::class)
    fun solve(input: BufferedReader): String {
        val n = input.readLine().toInt()
        val result = StringBuilder()
        val stack = Stack()
        for (i in 0 until n) {
            executeOperation(input.readLine(), stack, result)
        }
        return result.toString().trimEnd()
    }

    private fun executeOperation(
        operation: String,
        stack: Stack,
        result: StringBuilder
    ) {
        val st = StringTokenizer(operation, " ")
        when (st.nextToken()) {
            "push" -> stack.push(st.nextToken().toInt())
            "pop" -> result.append(stack.pop()).append("\n")
            "size" -> result.append(stack.size()).append("\n")
            "empty" -> result.append(stack.empty()).append("\n")
            "top" -> result.append(stack.top()).append("\n")
        }
    }

    internal class Stack {
        private var arr = intArrayOf()
        private var size = 0

        fun push(data: Int) {
            arr = arr.copyOf(size + 1)
            arr[size] = data
            size++
        }

        fun pop(): Int {
            if (size == 0) {
                return -1
            }
            val ret = arr[size - 1]
            arr[size - 1] = 0
            size--
            return ret
        }

        fun size(): Int {
            return size
        }

        fun empty(): Int {
            if (size == 0) {
                return 1
            }
            return 0
        }

        fun top(): Int {
            if (size == 0) {
                return -1
            }
            return arr[size - 1]
        }
    }
}
